api/osb/osb_controller.go

```diff
@@ -162,13 +169,25 @@ func (c *Controller) proxy(r *web.Request, logger *logrus.Entry, broker *types.S
 	return resp, nil
 }
 
-func buildProxy(targetBrokerURL *url.URL, logger *logrus.Entry, broker *types.ServiceBroker) *httputil.ReverseProxy {
+func buildProxy(targetBrokerURL *url.URL, logger *logrus.Entry, broker *types.ServiceBroker) (*httputil.ReverseProxy, error) {
 	proxy := httputil.NewSingleHostReverseProxy(targetBrokerURL)
 	director := proxy.Director
 	proxy.Director = func(request *http.Request) {
 		director(request)
 		logger.Infof("Forwarded OSB request to service broker %s at %s", broker.Name, request.URL)
 	}
+
+	brokerClient, err := client.New(broker, nil)
```

Comment:
`brokerClient` is not fully configured, `brokerClient.SendRequest` would crash

Reply:
The brokerClient sendRequest won't be used in this flow, we only need the client here to obtain a copy from the GetTransportWithTLS method.

Reply:
- Adjusted to use a broker-client transport object in the OSB, sm as a platform flows
- Client object will validate the func handler is not nil on New